api.go

```diff
@@ -133,8 +133,13 @@ func (h *Headscale) RegistrationHandler(c *gin.Context) {
 			Str("handler", "Registration").
 			Str("machine", m.Name).
 			Msg("Not registered and not NodeKey rotation. Sending a authurl to register")
-		resp.AuthURL = fmt.Sprintf("%s/register?key=%s",
-			h.cfg.ServerURL, mKey.HexString())
+
+		if h.cfg.OIDCIssuer != "" {
+			resp.AuthURL = fmt.Sprintf("%s/oidc/register/%s", h.cfg.ServerURL, mKey.HexString())
+		} else {
+			resp.AuthURL = fmt.Sprintf("%s/register?key=%s",
+				h.cfg.ServerURL, mKey.HexString())
+		}
```

Comment:
```suggestion
		if h.cfg.OIDCIssuer != "" {
			resp.AuthURL = fmt.Sprintf("%s/oidc/register/%s", strings.TrimSuffix(h.cfg.ServerURL, "/"), mKey.HexString())
		} else {
			resp.AuthURL = fmt.Sprintf("%s/register?key=%s",
				strings.TrimSuffix(h.cfg.ServerURL, "/"), mKey.HexString())
		}
```

Lets handle the possible scenario of users having a serverUrl with a trailing `/`